backend/api/EventHandlers/IsarConnectionEventHandler.cs

```diff
@@ -94,7 +94,11 @@ private void AddTimerForRobot(IsarRobotHeartbeatMessage isarRobotHeartbeat, Robo
             timer.Start();
 
             if (_isarConnectionTimers.TryAdd(robot.IsarId, timer)) { _logger.LogInformation("Added new timer for ISAR '{IsarId}' ('{RobotName}')", robot.IsarId, robot.Name); }
-            else { _logger.LogWarning("Failed to add new timer for ISAR '{IsarId}' ('{RobotName})'", robot.IsarId, robot.Name); }
+            else
+            {
+                _logger.LogWarning("Failed to add new timer for ISAR '{IsarId}' ('{RobotName})'", robot.IsarId, robot.Name);
+                timer.Close();
```

Comment:
How does the timer close for the if statement above

Reply:
Closes on the timeout event.